ethereum/contracts/PolkadotRelayChainBridge.sol

```diff
@@ -0,0 +1,296 @@
+         * the signature of senderPublicKey on the payload
+         */
+        // require(
+        //     ECDSA.recover(payload, validatorSignature) == validatorPublicKey,
+        //     "Error: Invalid Signature"
+        // );
+
+        // /**
+        //  * @dev Check that the bitfield actually contains enough claims to be succesful, ie, > 2/3
+        //  */
+        // require(
+        //     validatorClaimsBitfield.countSetBits() >
+        //         (validatorRegistry.numOfValidators() * THRESHOLD_NOM) /
+        //             THRESHOLD_DENOM,
+        //     "Error: Bitfield not enough validators"
+        // );
+
+        /**
+         * @todo Lock up the sender stake as collateral
+         */
+        // TODO
+
+        // Accept and save the commitment
+        validationData[currentId] = ValidationData(
+            msg.sender,
+            payload,
+            validatorClaimsBitfield,
+            block.number
+        );
+
+        emit InitialVerificationSuccessful(msg.sender, block.number, currentId);
+
+        currentId = currentId.add(1);
+    }
+
+    /**
+     * @notice Performs the second step in the validation logic
+     * @param id an identifying value generated in the previous transaction
+     * @param payload contains the payload signed by the validator(s)
+     * @param signatures an array of signatures from the randomly chosen validators
+     * @param validatorPositionsBitfield an array of bitfields from the chosen validators
+     * @param validatorPublicKeys an array of the public key of each signer
+     * @param validatorPublicKeyMerkleProofs an array of merkle proofs from the chosen validators
+     */
+    function completeSignatureCommitment(
+        uint256 id,
+        bytes32 payload,
+        bytes[] memory signatures,
+        uint256[] memory validatorPositionsBitfield,
+        address[] memory validatorPublicKeys,
+        bytes32[][] memory validatorPublicKeyMerkleProofs
+    ) public {
+        ValidationData storage data = validationData[id];
+
+        // TODO verify that sender is the same as in `newSignatureCommitment`
+        require(
+            msg.sender == data.senderAddress,
+            "Error: Sender address does not match original validation data"
+        );
+
+        // TODO calculate number of required validator signatures properly, eg:
+        //  uint8 numberOfValidators = validatorRegistry.numberOfValidators;
+        //  requiredNumberOfSamples = numberOfValidators * 2/3
+        uint8 requiredNumberOfSamples = 2;
+        /**
+         * @dev Generate an array of numbers
+         */
+        // uint8[] memory randomNumbers = getRandomNumbers(data, requiredNumberOfSamples);
+
+        /**
+         *  @dev For each randomSignature, do:
+         */
+        for (uint256 i = 0; i < requiredNumberOfSamples; i++) {
+            // @note Require random numbers generated onchain match random numbers
```

Comment:
The plan is to address the contract changes in a follow-on PR that integrates the latest Solidity changes from https://github.com/Snowfork/verification-solidity/pull/16. The contracts in this PR were pulled from prior versions and are essentially mocks to test the Relayer against.

Reply:
yeh i think we can do a cleanup PR after - so long as code from this PR actually works properly and is fit for staging, some mess that we cleanup straight after is fine

Reply:
Updated contracts to latest in https://github.com/Snowfork/polkadot-ethereum/pull/314/commits/79220513a46d60de6598935a6b1bda1ca3215d80 and then commented out contract verification checks in https://github.com/Snowfork/polkadot-ethereum/pull/314/commits/c55ba8d602d6122c4d8f2f0c185d5a6513fce8c7 that would cause the end-to-end process to fail.

Uncommenting the verification checks will cause the end-to-end BEEFY relay to fail at step #3 because the initial verification event won't be emitted. The idea is that once this PR is merged we immediately integrate the verification checks back in one by one.